src/Psr7/ControllerHelperTrait.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace MPScholten\RequestParser\Psr7;
+
+use MPScholten\RequestParser\RequestParser;
+use MPScholten\RequestParser\TypeParser;
+use Psr\Http\Message\ServerRequestInterface;
+
+trait ControllerHelperTrait
```

Comment:
Very good point.

We could introduce a `BaseControllerHelperTrait`:
```php
trait BaseControllerHelperTrait
{
    private $queryParser, $bodyParser;

    protected final function initRequestParser($request, callable $exceptionFactory = null)
    {
        $requestParser = $this->createRequestParser($request, $exceptionFactory);
        ...
    }
    protected function queryParameter($name) { ... }
    protected function bodyParameter($name) { ... }
}
```

And then have specific `Symfony\ControlerHelperTrait` and `Psr7\ControllerHelperTrait` implementations:

```php
trait SymfonyControllerTrait
{
    use BaseControllerHelperTrait;
    final protected function createRequestParser($request, $exceptionFactory)
    {
        return new SymfonyRequestParser($request, $exceptionFactory);
    }
}
```

What do you think?